Core/Src/main.c

```diff
@@ -0,0 +1,595 @@
+  * @file           : main.c
+  * @brief          : Main program body
+  ******************************************************************************
+  * @attention
+  *
+  * Copyright (c) 2023 STMicroelectronics.
+  * All rights reserved.
+  *
+  * This software is licensed under terms that can be found in the LICENSE file
+  * in the root directory of this software component.
+  * If no LICENSE file comes with this software, it is provided AS-IS.
+  *
+  ******************************************************************************
+  */
+/* USER CODE END Header */
+/* Includes ------------------------------------------------------------------*/
+#include "main.h"
+
+/* Private includes ----------------------------------------------------------*/
+/* USER CODE BEGIN Includes */
+#define DUTY_CYCLE 2000 //unused, can prolly delete
+/* USER CODE END Includes */
+
+/* Private typedef -----------------------------------------------------------*/
+/* USER CODE BEGIN PTD */
+
+/* USER CODE END PTD */
+
+/* Private define ------------------------------------------------------------*/
+/* USER CODE BEGIN PD */
+
+/* USER CODE END PD */
+
+/* Private macro -------------------------------------------------------------*/
+/* USER CODE BEGIN PM */
+
+/* USER CODE END PM */
+
+/* Private variables ---------------------------------------------------------*/
+TIM_HandleTypeDef htim2;
+
+UART_HandleTypeDef huart2;
+
+/* USER CODE BEGIN PV */
+
+/* USER CODE END PV */
+
+/* Private function prototypes -----------------------------------------------*/
+void SystemClock_Config(void);
+static void MX_GPIO_Init(void);
+static void MX_USART2_UART_Init(void);
+static void MX_TIM2_Init(void);
+/* USER CODE BEGIN PFP */
+
+/* USER CODE END PFP */
+
+/* Private user code ---------------------------------------------------------*/
+/* USER CODE BEGIN 0 */
+
+//feedback variables, used with interrupt handler to get incremental feedback from motor(s)
+int duration;
+int Direction;
+int YellowsigLast;
+
+int errorterm; //The difference between the current position and the desired position.
+
+//PID Control terms, adjust as needed
+int PropTerm = 10; //proportional term (30?)
+float IntTerm = 0; //Integral term
+float DerivTerm = 0.1; //Derivative term
+
+int FeedbackControlSignal; //final summation of effects of P, I and D controllers
+
+//Proportional control support variables
+int Pterm;
+
+//Integral control support variables
+int IntArray[30]; //= {0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
+int IntegralTerm;
+int ITerm;
+int sum;
+
+//Derivative control support variables
+int DerArray[30]; // = {1, 1, 1, 1, 1, 1, 1, 1, 1, 1};
+int DerivativeTerm;
+int DTerm;
+int numerator;
+int denominator;
+int slope;
+
+
+int DesiredPos; //desired final position of the actuator
+
+int GoalTerm = 1; //sometimes overshoots, leading to undefined behavior. This prevents movement once goal is reached.
+int i;
+int j;
+int k;
+int f;
+int z;
```

Comment:
What are these? Don't use variables that are only one letter. Always name variables so that someone unfamiliar with your code or project can understand immediately what is going on.